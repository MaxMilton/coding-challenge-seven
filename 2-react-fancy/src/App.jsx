import React, { Component } from 'react';
import { injectGlobal } from 'styled-components';
import Banner from './Banner';

/* eslint-disable-next-line no-unused-expressions */
injectGlobal`
  body {
    margin: 6rem;
    background-color: rgb(0, 0, 0);
  }
`;

export default class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      bannerTileX1: '../tile.jpg',
      bannerTileX2: '../tile@x2.jpg',
      bannerTitle: 'Home And Away',
    };
  }

  render() {
    const {
      bannerTileX1,
      bannerTileX2,
      bannerTitle,
    } = this.state;

    return (
      <Banner
        tileX1={bannerTileX1}
        tileX2={bannerTileX2}
        title={bannerTitle}
      />
    );
  }
}
