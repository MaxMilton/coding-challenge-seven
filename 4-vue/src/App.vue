<template>
  <div id="app">
    <Banner tile="../tile.jpg" title="Home And Away" />
  </div>
</template>

<script>
  import Banner from './Banner.vue';

  export default {
    components: {
      Banner,
    },
  };
</script>

<style>
  body {
    margin: 6rem;
    background-color: rgb(0, 0, 0);
  }
</style>
