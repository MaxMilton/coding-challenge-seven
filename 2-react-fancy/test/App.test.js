import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

describe('App root component', () => {
  it('renders correctly', () => {
    expect.assertions(2);
    const spy = jest.spyOn(console, 'error');
    const component = renderer.create(
      <App />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(spy).not.toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });
});
