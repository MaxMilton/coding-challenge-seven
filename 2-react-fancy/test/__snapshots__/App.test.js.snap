// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App root component renders correctly 1`] = `
<div
  className="sc-bdVaJa jCeSos"
>
  <picture>
    <img
      alt="Home And Away"
      className="sc-bwzfXH gbieAY"
      src="../tile@x2.jpg"
      srcSet="
        ../tile.jpg,
        ../tile@x2.jpg 2x
      "
    />
  </picture>
  <div
    className="sc-htpNat kuzZIM"
  >
    <picture>
      <img
        alt="7plus"
        className="sc-bxivhb kufJPE"
        src="../logo@x2.png"
        srcSet="
        ../logo.png,
        ../logo@x2.png 2x
      "
      />
    </picture>
    <h3
      className="sc-ifAKCX eVJwop"
    >
      Home And Away
    </h3>
  </div>
</div>
`;
